migratron/wait_for_hive.sh

```diff
@@ -0,0 +1,35 @@
+function wait_for_it() {
+    local serviceport=$1
+    local service=${serviceport%%:*}
+    local port=${serviceport#*:}
+    local retry_seconds=5
+    local max_try=100
+    let i=1
+
+    nc -z $service $port
+    result=$?
+
+    until [ $result -eq 0 ]; do
+        echo "[$i/$max_try] check for ${service}:${port}..."
+        echo "[$i/$max_try] ${service}:${port} is not available yet"
+        if (( $i == $max_try )); then
+            echo "[$i/$max_try] ${service}:${port} is still not available; giving up after ${max_try} tries. :/"
+            exit 1
+        fi
+      
+        echo "[$i/$max_try] try in ${retry_seconds}s once again ..."
+        let "i++"
+        sleep $retry_seconds
+
+        nc -z $service $port
+        result=$?
+    done
+    echo "[$i/$max_try] $service:${port} is available."
+}
```

Comment:
This function is super generic and I'm not sure if it serves our purpose. I would change it to:

```suggestion
function wait_for_beeline() {
    local retry_seconds=5
    local max_try=100
    let i=1

    beeline -u jdbc:hive2://hive-server:10000/ 'select 1'
    result=$?

    until [ $result -eq 0 ]; do
        echo "[$i/$max_try] check for beeline..."
        echo "[$i/$max_try] beeline is not available yet"
        if (( $i == $max_try )); then
            echo "[$i/$max_try] beeline is still not available; giving up after ${max_try} tries. :/"
            exit 1
        fi
      
        echo "[$i/$max_try] try in ${retry_seconds}s once again ..."
        let "i++"
        sleep $retry_seconds

        beeline -u jdbc:hive2://hive-server:10000/ 'select 1'
        result=$?
    done
    echo "[$i/$max_try] beeline is available."
}
```

This checks that beeline is not only accesible, but it is also queryable